Src/Base/AMReX_BoxArray.cpp

```diff
@@ -1275,99 +1275,90 @@ BoxArray::complementIn (BoxList& bl, const Box& bx) const
-                        const Box& isect = bx & ibox;
-
-                        if (isect.ok())
-                        {
-                            newbl.clear();
-                            for (const Box& b : bl) {
-                                amrex::boxDiff(newdiff, b, isect);
-                                newbl.join(newdiff);
-                            }
-                            bl.swap(newbl);
-                        }
+            amrex::ignore_unused(j,k);
+            auto it = BoxHashMap.find(IntVect(AMREX_D_DECL(i,j,k)));
+            if (it != TheEnd) {
+                for (const int index : it->second) {
+                    const Box& ibox = abox[index];
+                    if (bx.intersects(ibox)) {
+                        intersect_boxes.push_back(ibox);
                     }
-                } else if (m_bat.is_simple()) {
-                    IndexType t = ixType();
-                    IntVect cr = crseRatio();
-                    for (const int index : it->second)
-                    {
-                        const Box& ibox = amrex::convert(amrex::coarsen(abox[index],cr),t);
-                        const Box& isect = bx & ibox;
-
-                        if (isect.ok())
-                        {
-                            newbl.clear();
-                            for (const Box& b : bl) {
-                                amrex::boxDiff(newdiff, b, isect);
-                                newbl.join(newdiff);
-                            }
-                            bl.swap(newbl);
-                        }
+                }
+            }
+        });
+    } else if (m_bat.is_simple()) {
+        IndexType t = ixType();
+        IntVect cr = crseRatio();
+        AMREX_LOOP_3D(cbx, i, j, k,
+        {
+            amrex::ignore_unused(j,k);
+            auto it = BoxHashMap.find(IntVect(AMREX_D_DECL(i,j,k)));
+            if (it != TheEnd) {
+                for (const int index : it->second) {
+                    const Box& ibox = amrex::convert(amrex::coarsen(abox[index],cr),t);
+                    if (bx.intersects(ibox)) {
+                        intersect_boxes.push_back(ibox);
                     }
-                } else {
-                    for (const int index : it->second)
-                    {
-                        const Box& ibox = m_bat.m_op.m_bndryReg(abox[index]);
-                        const Box& isect = bx & ibox;
-
-                        if (isect.ok())
-                        {
-                            newbl.clear();
-                            for (const Box& b : bl) {
-                                amrex::boxDiff(newdiff, b, isect);
-                                newbl.join(newdiff);
-                            }
-                            bl.swap(newbl);
-                        }
+                }
+            }
+        });
+    } else {
+        AMREX_LOOP_3D(cbx, i, j, k,
+        {
+            amrex::ignore_unused(j,k);
```

Comment:
could this be incorporated into the `AMREX_LOOP_3D` macro somehow?

Reply:
We certainly could.

Reply:
I will do this in a new PR.